### PR TITLE
🍼 Google button on YPP

### PR DESCRIPTION
**Pull request by WRadoslaw**
#4088

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,5 +82,8 @@
   "engines": {
     "node": "^16"
   },
-  "packageManager": "yarn@3.2.1"
+  "packageManager": "yarn@3.2.1",
+  "dependencies": {
+    "react-google-button": "^0.7.2"
+  }
```
Comment (drillprop) [inline package.json]: why didn't you install it only for the `atlas` package? Why do we need it here?

File: packages/atlas/src/assets/google/googleLogo.webp
Comment (drillprop) [inline packages/atlas/src/assets/google/googleLogo.webp:1]: Like I said in the standup. I think we could use SVG here. You can add it either by yourself in the codebase or ask Kuba to add them to Figma - somewhere here https://www.figma.com/file/1uxugLps6oooI38PGmdrn9/Figma-API--%3E-Storybook?node-id=637-379&t=29guQf48Bguay3R0-4 or in illustrations.

The latter would be better, but I'm fine if you just add this to assets.

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx [hunk 2]
```diff
@@ -152,7 +157,7 @@ export const YppAuthorizationModal: FC<YppAuthorizationModalProps> = ({
     }
     if (currentStep === 'requirements' && hasMoreThanOneChannel) {
       setYtRequirementsErrors([])
-      onChangeStep('select-channel')
+      // onChangeStep('select-channel')
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx]: to remove or to keep it?

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx [hunk 3]
```diff
@@ -326,66 +331,65 @@ export const YppAuthorizationModal: FC<YppAuthorizationModalProps> = ({
...
-    setSelectedChannelId,
-  ])
-
   const authorizationStep = useMemo(() => {
     switch (currentStep) {
-      case 'requirements':
+      case 'requirements': {
+        const getPrimaryButton = () => {
+          if (isLoading) {
+            return <Spinner />
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx]: why do we use the Spinner now? Previously we had the button with the text `'Please wait...'` and this is what we have in other places in the app, so for the sake of consistency, I would use regular button

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationSteps/YppAuthorizationRequirementsStep/YppAuthorizationRequirementsStep.tsx [hunk 4]
```diff
@@ -15,15 +16,18 @@ type YppAuthorizationRequirementsStepProps = {
...
+        {requirements.map((item, idx) => (
+          <ListItem key={idx} as="li" variant="t200" color="colorText">
+            <TickWrapper fulfilled={item.fulfilled}>
+              {item.fulfilled ? <SvgActionCheck /> : <SvgActionClose />}
+            </TickWrapper>
+            <Paragraph>{item.text}</Paragraph>
+          </ListItem>
+        ))}
+      </StyledList>
+      <Banner description="Gleev uses Google OAuth to get access to your public profile and account email address as part of sign up flow, and integrates with YouTube API to obtain details about your YouTube channel data, such as followers and video statistics." />
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationSteps/YppAuthorizationRequirementsStep/YppAuthorizationRequirementsStep.tsx]: we should use here appName

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx (hunk 3 above)
Comment (WRadoslaw) [reply packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx]: The problem is that we used to display "please wait", cause only the text changed back then. Imho using button as a loading state when in the end button can be switched to Google one is weird. I would go either with a spinner, skeleton or disabled textbutton

File: packages/atlas/src/components/_buttons/GoogleButton/GoogleButton.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+import { SvgGoogleSmallLogo } from '@/assets/icons/GoogleSmallLogo'
+
+interface GoogleButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
+  label?: string
+  disabled?: boolean
+}
+export const GoogleButton = ({ label, ...rest }: GoogleButtonProps) => {
+  return (
+    <Container as="button" {...rest}>
```
Comment (drillprop) [inline packages/atlas/src/components/_buttons/GoogleButton/GoogleButton.tsx]: This is a button, we don't need `as` here
```suggestion
    <Container {...rest}>
```

File: packages/atlas/src/components/_buttons/GoogleButton/GoogleButton.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+const IconContainer = styled.div`
+  background-color: #fff;
+  height: 100%;
+  display: grid;
+  place-items: center;
+`
+
+const Container = styled.button`
```
Comment (drillprop) [inline packages/atlas/src/components/_buttons/GoogleButton/GoogleButton.tsx]: I would name this StyledGoogleButton or something similar with a name that contains the word `button`. It's not a container

File: packages/atlas/src/components/_buttons/GoogleButton/GoogleButton.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  display: inline-flex;
+  align-items: center;
+  text-align: center;
+  padding: 0;
+  box-shadow: 0 2px 4px 0 rgb(0 0 0 / 0.25);
+  font-size: 16px;
+  border-radius: 1px;
+  transition: background-color 0.218s, border-color 0.218s, box-shadow 0.218s;
+  font-family: Roboto, arial, sans-serif;
```
Comment (drillprop) [inline packages/atlas/src/components/_buttons/GoogleButton/GoogleButton.tsx:37]: We don't have available Roboto font in the app, so the rendered font will be Arial: 
image
I'm a little worried that Google team could not accept this

**Review by drillprop — DISMISSED** [review]

**Review by attemka — APPROVED** [review]

**Pull request merged**